### PR TITLE
Activate root elements for activate instructions of PI modification

### LINKED ISSUE #9640: Modify a process instance by activating root elements
## Description

We want to support the basic modification by activating root elements (i.e. elements that are not embedded in a subprocess). We want to create a new element instance for each activate instruction. The element instances should be created by writing an `activate` command for the given elements. As a result, the process instance should continue at the given elements.

For example:
image

* write an `activate` command for each activate instruction
* verify that 
  * the process instance activates the given elements
  * the process instance can complete the given elements
  * the process instance is completed when the given elements and the other element instances are completed

Out of scope:
* nested elements inside subprocesses
* not supported elements (see #9729)

Blocked by #9639

### PR DESCRIPTION
**Pull request by korthout**
## Description

This adds the ability to the process instance modification `MODIFY` command processor to activate root process elements when these are specified in the activate instructions.

It also provides the basic test harness required to test this, and that is also needed for the upcoming issues.

Lastly, this PR rectifies some details about the process instance modification `MODIFIED` event that is written as a result of processing the `MODIFY` command.

## Related issues

closes #9640 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by korthout** [comment]
I force pushed to fix a missing license header in one of the commits.

**Comment by korthout** [comment]
Interesting. This flaky test should've been fixed with 365f35a5bcd1d489607d7ca287aafef37e9a46a2, which is an ancestor of this branch `korthout-9640-activate-root-elements`. I'll reopen it.
- #9813

**Review by saig0 — CHANGES_REQUESTED** [review]
@korthout great job :1st_place_medal: Clean code and easy to review - thanks to the separated commits and commit messages :rocket: 

I miss only one test case. Please see my comment. Then, it's ready to merge :heavy_check_mark:

File: engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTest.java [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        .hasIntent(ProcessInstanceModificationIntent.MODIFIED);
+
+    assertThat(event.getValue())
+        .hasProcessInstanceKey(processInstanceKey)
+        .hasNoActivateInstructions()
+        .hasNoTerminateInstructions();
+  }
+
+  @Test
+  public void shouldActivateRootElement() {
```
[the remark this answers is not in the document]
Comment (korthout) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTest.java:69]: @saig0 I've added a test. Would you've also asserted the exact data in the records, or do you think it's enough to assert the activation of the elements?

[the remark this answers is not in the document]
Comment (saig0) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTest.java:69]: I think it is good as it is now :+1:

**Review by saig0 — APPROVED** [review]
@korthout LGTM :rocket:

**Comment by korthout** [comment]
bors merge

**Pull request merged**